### PR TITLE
Add `EventSender` to resolve dependency cycle

### PR DESCRIPTION
**Pull request by pinkwah**
`Scheduler` has a reference to `Job` and `Job` has a reference to `Scheduler`. Adding `EventSender` lets us resolve this cycle as now `Scheduler` has a reference to `Job`s and `EventSender`, but each `Job` only refers to `EventSender`.

### CODE REVIEW

**Comment by xjules** [comment]
This needs rebasing first

**Comment by berland** [comment]
Can this solve https://github.com/equinor/ert/blob/881a350c2fef60983a1042309d82ad39aeac8178/src/ert/scheduler/job.py#L28-L29 ?

File: src/ert/scheduler/scheduler.py [hunk 1]
```diff
@@ -57,38 +63,32 @@ def __init__(
         self.driver = driver
         self._tasks: MutableMapping[int, asyncio.Task[None]] = {}
 
-        self._jobs: MutableMapping[int, Job] = {
-            real.iens: Job(self, real) for real in (realizations or [])
+        self._jobs: Mapping[int, Job] = {
+            real.iens: Job(real, on_complete=self._on_job_complete)
```
Comment (xjules) [inline src/ert/scheduler/scheduler.py:67]: Looks good! But (just our of curiosity mostly), why not to to use:
`self._tasks[iens].add_done_callback(self._on_job_complete)`?

Comment (pinkwah) [reply src/ert/scheduler/scheduler.py:67]: Because it's not when the task is done running, it's when the job is `COMPLETED` successfully.

Comment (xjules) [reply src/ert/scheduler/scheduler.py:67]: Yes, but we can check in `_on_job_complete` that `self.state == COMPLETED`?

File: src/ert/scheduler/event_sender.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            ssl=tls,
+            extra_headers=headers,
+            open_timeout=60,
+            ping_timeout=60,
+            ping_interval=60,
+            close_timeout=60,
+        ):
+            while True:
+                event = await self.events.get()
+                print(f"==SENDING {event=}")
```
Comment (berland) [inline src/ert/scheduler/event_sender.py:68]: leftover?

File: src/ert/scheduler/job.py [hunk 3]
```diff
@@ -87,13 +95,15 @@ def running_duration(self) -> float:
             return time.time() - self._start_time
         return 0
 
-    async def _submit_and_run_once(self, sem: asyncio.BoundedSemaphore) -> None:
+    async def _submit_and_run_once(
+        self, sem: asyncio.BoundedSemaphore, driver: Driver
+    ) -> None:
```
Comment (berland) [inline src/ert/scheduler/job.py:100]: is this driver arg related to the commit message, or is it some other refactoring?

Comment (pinkwah) [reply src/ert/scheduler/job.py:100]: It's part of the commit. The `Job` object obtains a `Driver` only when it is supposed to use it, not before.

**Comment by pinkwah** [comment]
Closing as this doesn't really solve anything. There will be a cyclic dependency regardless.

**Pull request closed without merging**